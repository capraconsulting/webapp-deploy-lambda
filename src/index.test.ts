import * as iam from "@aws-cdk/aws-iam"
import * as s3 from "@aws-cdk/aws-s3"
import * as cdk from "@aws-cdk/core"
import "jest-cdk-snapshot"
import { WebappDeploy } from "."

test("WebappDeploy", () => {
  const app = new cdk.App()
  const stack1 = new cdk.Stack(app, "Stack1", {
    env: {
      account: "123456789123",
      region: "us-east-1",
    },
  })
  const stack2 = new cdk.Stack(app, "Stack2", {
    env: {
      account: "112233445566",
      region: "eu-west-1",
    },
  })

  const buildsBucket = new s3.Bucket(stack1, "BuildsBucket", {
    bucketName: cdk.PhysicalName.GENERATE_IF_NEEDED,
  })
  const webBucket = new s3.Bucket(stack1, "WebBucket", {
    bucketName: cdk.PhysicalName.GENERATE_IF_NEEDED,
  })

  const callerRole = new iam.Role(stack2, "CallerRole", {
    assumedBy: new iam.AnyPrincipal(),
  })

  const webappDeploy = new WebappDeploy(stack2, "WebappDeploy", {
    buildsBucket,
    distributionId: "EKJ2IPY1KTEAR1",
    webBucket,
  })

  webappDeploy.deployFn.grantInvoke(callerRole)

  expect(stack2).toMatchCdkSnapshot()
})
