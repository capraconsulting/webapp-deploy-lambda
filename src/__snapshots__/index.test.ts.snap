// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebappDeploy 1`] = `
Object {
  "Parameters": Object {
    "AssetParameterscd77cffef8c2f4b055f0e3fbe483471091af3140ee9aaba99cb061e3833caa86ArtifactHashC97E7695": Object {
      "Description": "Artifact hash for asset \\"cd77cffef8c2f4b055f0e3fbe483471091af3140ee9aaba99cb061e3833caa86\\"",
      "Type": "String",
    },
    "AssetParameterscd77cffef8c2f4b055f0e3fbe483471091af3140ee9aaba99cb061e3833caa86S3Bucket5B916101": Object {
      "Description": "S3 bucket for asset \\"cd77cffef8c2f4b055f0e3fbe483471091af3140ee9aaba99cb061e3833caa86\\"",
      "Type": "String",
    },
    "AssetParameterscd77cffef8c2f4b055f0e3fbe483471091af3140ee9aaba99cb061e3833caa86S3VersionKey8C593C94": Object {
      "Description": "S3 key for asset version \\"cd77cffef8c2f4b055f0e3fbe483471091af3140ee9aaba99cb061e3833caa86\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "CallerRole7E4BBD59": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": "*",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "CallerRoleDefaultPolicyECDC9714": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "WebappDeploy4D661524",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CallerRoleDefaultPolicyECDC9714",
        "Roles": Array [
          Object {
            "Ref": "CallerRole7E4BBD59",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "WebappDeploy4D661524": Object {
      "DependsOn": Array [
        "WebappDeployServiceRoleDefaultPolicyE446530F",
        "WebappDeployServiceRole78F79841",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameterscd77cffef8c2f4b055f0e3fbe483471091af3140ee9aaba99cb061e3833caa86S3Bucket5B916101",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameterscd77cffef8c2f4b055f0e3fbe483471091af3140ee9aaba99cb061e3833caa86S3VersionKey8C593C94",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameterscd77cffef8c2f4b055f0e3fbe483471091af3140ee9aaba99cb061e3833caa86S3VersionKey8C593C94",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "CF_DISTRIBUTION_ID": "EKJ2IPY1KTEAR1",
            "DEPLOY_LOG_BUCKET_URL": "s3://stack1stack1webbucket4a67ef3a90b336dbc074/deployments.log",
            "EXPIRE_SECONDS": "432000",
            "TARGET_BUCKET_URL": "s3://stack1stack1webbucket4a67ef3a90b336dbc074/web",
          },
        },
        "Handler": "webapp_deploy.main.handler",
        "ReservedConcurrentExecutions": 1,
        "Role": Object {
          "Fn::GetAtt": Array [
            "WebappDeployServiceRole78F79841",
            "Arn",
          ],
        },
        "Runtime": "python3.7",
        "Timeout": 120,
      },
      "Type": "AWS::Lambda::Function",
    },
    "WebappDeployServiceRole78F79841": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "WebappDeployServiceRoleDefaultPolicyE446530F": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:HeadObject",
                "s3:GetObject",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":s3:::stack1ack1buildsbucket2655d1d2c6f4cd55937f/*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "s3:PutObject",
                "s3:DeleteObject",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":s3:::stack1stack1webbucket4a67ef3a90b336dbc074/web/*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "s3:GetObject",
                "s3:PutObject",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":s3:::stack1stack1webbucket4a67ef3a90b336dbc074/deployments.log",
                  ],
                ],
              },
            },
            Object {
              "Action": "cloudfront:CreateInvalidation",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "WebappDeployServiceRoleDefaultPolicyE446530F",
        "Roles": Array [
          Object {
            "Ref": "WebappDeployServiceRole78F79841",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
